src/main/java/org/javarosa/core/services/storage/StorageManager.java

```diff
@@ -45,6 +45,13 @@ public void registerStorage(String key, Class type) {
         if (storageFactory == null) {
             throw new RuntimeException("No storage factory has been set; I don't know what kind of storage utility to create. Either set a storage factory, or register your StorageUtilitys directly.");
         }
+        if (storageRegistry.containsKey(key)) {
+            if (storageRegistry.get(key).getClass() == type) {
+                return;
+            } else {
+                Logger.log("Warning", "Overwriting storage type " + storageRegistry.get(key).getClass() + " with type " + type + " for key " + key + ".");
```

Comment:
probably throw a `RuntimeException` from here ? I would think it almost always a code mistake if the storage is keyed with 2 different types in the same request lifecycle.

Reply:
+1 for throwing an exception here.